test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -139,23 +141,65 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 			sc = volumeResource.Sc
 			claimSize = pvc.Spec.Resources.Requests.Storage().String()
 
-			ginkgo.By("starting a pod to use the claim")
+			ginkgo.By("[init] starting a pod to use the claim")
 			originalMntTestData = fmt.Sprintf("hello from %s namespace", pvc.GetNamespace())
 			command := fmt.Sprintf("echo '%s' > %s", originalMntTestData, datapath)
 
-			RunInPodWithVolume(cs, f.Timeouts, pvc.Namespace, pvc.Name, "pvc-snapshottable-tester", command, config.ClientNodeSelection)
+			pod := RunInPodWithVolume(cs, f.Timeouts, pvc.Namespace, pvc.Name, "pvc-snapshottable-tester", command, config.ClientNodeSelection)
 
 			err = e2epv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, cs, pvc.Namespace, pvc.Name, framework.Poll, f.Timeouts.ClaimProvision)
 			framework.ExpectNoError(err)
-			ginkgo.By("checking the claim")
+
 			// Get new copy of the claim
+			ginkgo.By("[init] checking the claim")
 			pvc, err = cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
 			framework.ExpectNoError(err)
 
 			// Get the bound PV
-			ginkgo.By("checking the PV")
-			_, err = cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+			ginkgo.By("[init] checking the PV")
+			pv, err := cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
 			framework.ExpectNoError(err)
+
+			// At this point we know that:
+			// - a pod was created with a PV that's supposed to have data
+			// However there's a caching issue that @jinxu97 explained and it's related with the pod & volume
+			// lifecycle in windows, to understand it we first analyze what the volumemanager does:
+			// - when a container is no longer running the volumemanager will try to cleanup the volume mounts
+			//   - Unpublish: unbinds the global mount from the container
```

Comment:
NodeUnpublishVolume: umount the bind mount (not the global mount)

Maybe simplify this comment: 
key point: 
Pod delete might not guarantee data flushed to disk, especially for Windows.  But NodeUnstageVolume adds the logic to flush the data to disk. See issue https://github.com/kubernetes/kubernetes/issues/81690 for details. 
So here by checking volumeInUse in node status, the test waits after volume is removed from volumeInUse list which indicating NodeUnstageVolume finishes before taking snapshot.

Reply:
I've updated the comments, please let me know if you think I missed something